File: src/main/java/de/tectoast/emolga/utils/records/DeferredSlashResponse.java
package de.tectoast.emolga.utils.records;

import net.dv8tion.jda.api.interactions.InteractionHook;

import java.util.concurrent.CompletableFuture;

public record DeferredSlashResponse(CompletableFuture<InteractionHook> cf) {

    public void reply(String message) {
        cf.thenCompose(ih -> ih.sendMessage(message).submit());
    }
}

File: src/main/kotlin/de/tectoast/emolga/utils/records/DeferredSlashResponse.kt
package de.tectoast.emolga.utils.records

import net.dv8tion.jda.api.interactions.InteractionHook
import java.util.concurrent.CompletableFuture

class DeferredSlashResponse(private val cf: CompletableFuture<InteractionHook>) {
    fun reply(message: String) {
        cf.thenCompose { ih: InteractionHook ->
            ih.sendMessage(
                message
            ).submit()
        }
    }
}